### DISCUSSION TITLE
Validation - Min & Numeric

### DISCUSSION THREAD

**Discussion by jynxy** [livewire/livewire#7977] [1. Help]
Hi,

Trying to set a input field so that is has a min of 6 characters long and must be numeric.

When i type numbers the first number shows its not long enough, soon as i enter the second number it shows as if it is long enough.

Weirdly if i enter letters, this works as expected soon as i get 6 or more characters all is ok, less than 6 and shows not long enough. when entering letters it shows that it is not numeric, so i know that part is working.

Any ideas why ?

    `#[Validate('required', message: 'Ticket number is required.')]
    #[Validate('min:6', message: 'Ticket number must be at least 6 characters.')]
    #[Validate('numeric', message: 'Ticket number must only contain numbers.')]
    public $ticket = '';`

`								<div class="row align-items-center">
									<div class="col-5">
										<label class="col-form-label @error('ticket') text-danger @enderror"><b>Ticket #</b></label>
									</div>
									<div class="col-7 ps-0">
										<input type="text" class="form-control form-control-sm @error('ticket') is-invalid @enderror" wire:model.blur="ticket" tabindex="3">
									</div>
								</div>`

Thanks

**Comment by joshhanley** [comment]
@jynxy I can't remember, but maybe having 3 separate `#Validate` calls is causing issues. Try merging them into one.

```php
#[Validate(['required', 'min:6', 'numeric'], attribute: 'Ticket number')]
```

By using `attribute` you don't need to write out each message, instead just change the label of the property.

Hope this helps!

**Comment by jynxy** [comment]
@joshhanley thanks for the reply much appreciated, unfortunately i still get the same result. any other ideas or is this a bug ?

**Reply by joshhanley** [reply]
@jynxy hmm strange. Can you make a https://wirebox.app that demonstrates the issue so we can have a look?

**Comment by shubdragon** [comment]
I think in case of numeric you need [min_digits:value](https://laravel.com/docs/10.x/validation#rule-min-digits) and [max_digits:value](https://laravel.com/docs/10.x/validation#rule-max-digits) instead of `max` and `min` 
because it is validating like this if number is less than 5 than it will show less than 6 not character in case of numeric value

**Comment by jynxy** [comment]
Sorry here is the wirebox, think i have made it public, was hidden away.

https://wirebox.app/b/4mpn5

**Reply by joshhanley** [reply]
@jynxy thanks! Yeah so min, is like if it's a number less than 6, then it will fail (try typing `5` in your wirebox, and then try `6`).

As mentioned above by @shubdragon, you need to use `min_digits` rule.

So try this

```php
#[Validate(['required', 'min_digits:6', 'numeric'], attribute: 'Ticket number')]
```

Hope this helps!

**Reply by jynxy** [reply]
Oh, erm, that's embarrassing, I assumed min would be the min length. Seems that works. Everyday is a school day. 

Im new to Laravel, so a lot to learn still. So i have been using min for everything else, is this correct or is there say a min_char ?

Thanks for your help.

**Reply by joshhanley** [reply]
@jynxy all good! We're always learning in this field 🙂 I use min for text characters (like if you type `abc` in your wirebox is throws the min:6 required error but if you type `abcdefg` it is happy). I think possibly the combination of that with the numeric rule requires the `min_digits` rule.
